opentelemetry/proto/metrics/v1/metrics.proto

```diff
@@ -138,6 +138,41 @@ message HistogramTimeSeries {
 
   // The data points of this timeseries.
   repeated HistogramValue points = 2;
+
+  // ExplicitBounds specifies buckets with explicitly defined bounds for values.
+  message ExplicitBounds {
+    // The bucket boundaries are described by "bounds" field.
+    //
+    // This defines size(bounds) + 1 (= N) buckets. The boundaries for bucket
+    // at index i are:
+    //
+    // [0, bounds[i]) for i == 0
+    // [bounds[i-1], bounds[i]) for 0 < i < N-1
+    // [bounds[i], +infinity) for i == N-1
+    // The values in bounds array must be strictly increasing and > 0.
+    //
+    // Note: only [a, b) intervals are currently supported for each bucket. If we decides
+    // to also support (a, b] intervals we should add support for these by defining a boolean
+    // value which decides what type of intervals to use.
+    repeated double bounds = 1;
+  }
+
+  // A histogram may optionally contain the distribution of the values in the population.
+  // In that case "bucket_options" field below and "buckets" field in HistogramValue
+  // both must be defined. Otherwise both fields must be omitted in which case the
+  // distribution of values in the histogram is unknown and only the total count and sum
+  // are known.
+  //
+  // bucket_options apply to all points in this HistogramTimeSeries. To define
+  // different bucket_options for different points create separate instances of
+  // HistogramTimeSeries each with its own bucket_options.
+  oneof bucket_options {
```

Comment:
Yes, that's a good idea unless we think we may have a large number of different bucket options (which would result in memoery size increase). I think we are likely only going to add Linear and Exponential. In that case oneof can be eliminated and we will save CPU time. Let me do a benchmark and post results.

Reply:
Benchmarked without oneof and 3 bucket options (with only one set). Memory consumption is virtually identical as with oneof but we gain a few more percent of CPU speed.

With single bucket option (onyl explicit_bounds, like we have now) the oneof version is even worse.

I will eliminate oneof since it is a clear improvement.

Reply:
I removed oneof and updated PR description with latest benchmark results.